src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -396,25 +374,115 @@ protected final boolean walkToBuilding()
     }
 
     /**
+     * Check all chests in the worker hut for a required item.
+     * @param is the type of item requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isInHut(@Nullable final ItemStack is)
+    {
+        @Nullable final AbstractBuildingWorker building = getOwnBuilding();
+
+        boolean hasItem;
+        if(building != null)
+        {
+            hasItem = isInTileEntity(building.getTileEntity(), is);
+
+            if(hasItem)
+            {
+                return true;
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
+            {
+                TileEntity entity = world.getTileEntity(pos);
+                if(entity instanceof TileEntityChest)
+                {
+                    hasItem = isInTileEntity((TileEntityChest) entity, is);
+
+                    if(hasItem)
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Check all chests in the worker hut for a required tool.
+     * @param tool the type of tool requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isToolInHut(final String tool)
+    {
+        @Nullable final AbstractBuildingWorker building = getOwnBuilding();
+
+        boolean hasItem;
+        if(building != null)
+        {
+            hasItem = isToolInTileEntity(building.getTileEntity(), tool);
+
+            if(hasItem)
+            {
+                return true;
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'pos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)